### PR TITLE
Make event handling asynchronous

### PR DESCRIPTION
**Pull request by alancutter**
When players are played in the end event of the last player they do not get played.
Making the events fire asynchronously fixes the problem by allowing player registration to occur at the appropriate time.

### CODE REVIEW

**Comment by alancutter** [comment]
The testharness can't handle two levels of asynchronousicity unfortunately so the test for this bug has to be manual.

File: test/testcases/manual-test-add-player-onend.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+<script src="../bootstrap.js"></script>
+<script>
+'use strict';
+
+var test = async_test('Check second animation has fired after being triggered inside onend callback');
+var element = document.querySelector('#target');
+var animation = new Animation(element, { left: '50px' }, 0.5);
+animation.addEventListener('end', function() {
+  console.log('testpants');
```
Comment (dstoc) [inline test/testcases/manual-test-add-player-onend.html]: Should this be 'testPants' for consistency?

Comment (dstoc) [inline test/testcases/manual-test-add-player-onend.html]: On second thoughts best not to have logging at all ;)

File: test/testcases/manual-test-add-player-onend.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+<div id="expected"></div>
+<div id="target"></div>
+
+<script src="../bootstrap.js"></script>
+<script>
+'use strict';
+
+var test = async_test('Check second animation has fired after being triggered inside onend callback');
+var element = document.querySelector('#target');
+var animation = new Animation(element, { left: '50px' }, 0.5);
```
Comment (dstoc) [inline test/testcases/manual-test-add-player-onend.html]: don't need spaces in a single line object initializer:

```
{left: '50px'}
```

File: test/testcases/manual-test-add-player-onend.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  nextAnimation.addEventListener('end', function() {
+    console.log('testChicken');
+    test.step(function() { assert_styles("#target", {'left':'100px'}); });
+    test.done();
+  });
+  document.timeline.play(nextAnimation);
+});
+document.timeline.play(animation);
+
+timing_test(function() {at(1, function() {});}, 'Force testharness to execute to 1 second.');
```
Comment (dstoc) [inline test/testcases/manual-test-add-player-onend.html:56]: Hmm, does this still not work when named unit-test-*.html ? I had assumed all of the timing test logic would be disable then.

Comment (alancutter) [inline test/testcases/manual-test-add-player-onend.html:56]: Tried as unit-test-*, still failed to execute second end event handler and timed out the test.

File: test/testcases/manual-test-add-player-onend.html (hunk 2 above)
Comment (alancutter) [inline test/testcases/manual-test-add-player-onend.html]: Done.

File: test/testcases/manual-test-add-player-onend.html (hunk 1 above)
Comment (alancutter) [inline test/testcases/manual-test-add-player-onend.html]: The non-camel cased debug print comes back to haunt me!
Yes it should be testPants and yes it should not be there. :)

File: test/testcases/manual-test-add-player-onend.html (hunk 3 above)
Comment (alancutter) [inline test/testcases/manual-test-add-player-onend.html:56]: Filed #426 for this issue.

**Pull request merged**